test/support/enterprise.ex

```diff
@@ -0,0 +1,143 @@
+defmodule LivebookTest.Enterprise do
```

Comment:
Currently we setup enterprise per case, which means they either need to be `async: false` or we need to stat multiple enterprises on different ports.

I'm thinking we could define `EnterpriseServer` (a named `GenServer`), and we would always start just one if needed. This way multiple suites can reuse the same enterprise instance.

We can then use the `EnterpriseServer` for all the management we need, like getting the token/url, etc. wdyt?

Reply:
This will also make it easier to consume the port output :)

Reply:
In this case, we would run all tests in `async: true` and using the same `GenServer` to handle the server integration, right?

Reply:
Yes!

Reply:
cool, let's do it